### PR TITLE
XW-5041 | Mobile / Modal search unit

### PR DESCRIPTION
**Pull request by bkoval**
@Wikia/x-wing

### CODE REVIEW

File: app/components/wds-global-navigation.js [hunk 1]
```diff
@@ -71,9 +71,17 @@ export default Component.extend({
 
 		openModal(modalType) {
 			this.set('currentModal', modalType);
+
+			if (modalType === 'search') {
+				this.send('activateSearch');
```
Comment (hakubo) [inline app/components/wds-global-navigation.js]: this.get('activateSearch')() - does this does not work here?

File: style-guide/styles/wds-components/_search-modal.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	.wds-global-navigation__search-submit {
+		display: none;
+	}
+
+	.wds-global-navigation__search-suggestions {
+		left: -12px;
+		right: -12px;
+	}
+
+	.wds-global-navigation__link-button[href *= "Special:CreateNewWiki"] {
```
Comment (hakubo) [inline style-guide/styles/wds-components/_search-modal.scss]: add a comment - why this is needed, please

File: app/components/wds-global-navigation.js [hunk 3]
```diff
@@ -62,7 +62,7 @@ export default Component.extend({
 		},
 
 		onSearchSuggestionChosen(suggestion) {
-			this.searchSuggestionChosen(suggestion);
+			this.get('onSearchSuggestionChosen')(suggestion);
```
Comment (latata) [inline app/components/wds-global-navigation.js]: `this.onSearchSuggestionChosen(suggestion)`

File: style-guide/styles/wds-components/_global-navigation-search.scss [hunk 4]
```diff
@@ -21,6 +21,18 @@ $toggle-active-margin-right: 17px;
 		transition: flex $search-transition-duration;
 	}
 
+	&-clear {
+		transform: rotate(45deg);
+
+		svg {
+			fill: white;
```
Comment (latata) [inline style-guide/styles/wds-components/_global-navigation-search.scss]: `$wds-color-white`

File: style-guide/styles/wds-components/_search-modal.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
+.wds-search-modal {
+	&__form-and-links-wrapper {
+		background-color: $wds-fandom-color-navy;
+		padding: 0 12px 12px;
+
+		.wds-global-navigation__links {
+			display: flex;
+			font-size: 12px;
+			justify-content: space-between;
+			margin: 30px 6px 0 6px;
```
Comment (latata) [inline style-guide/styles/wds-components/_search-modal.scss]: `margin: 30px 6px 0;`

File: style-guide/styles/wds-components/_search-modal.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+	&__search-icon {
+		fill: $wds-color-white;
+		margin-right: 12px;
+	}
+}
+
+.wds-search-is-active .wds-search-modal {
+	.wds-global-navigation__search-container {
+		border-top: 1px solid rgba(255, 255, 255, 0.25);
```
Comment (latata) [inline style-guide/styles/wds-components/_search-modal.scss]: `rgba(#fff, .25)`

File: app/templates/components/wds-global-navigation/search-modal.hbs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+	{{wds-icon name='magnifying-glass'}}
+{{/wds-button}}
+
+{{#if isOpen}}
+	<div class="wds-global-navigation__modal wds-search-modal">
+		<div class="wds-search-modal__form-and-links-wrapper">
+			{{wds-global-navigation/search
+				model=model
+				activateSearch=activateSearch
+				deactivateSearch=(action 'deactivateSearch')
```
Comment (latata) [inline app/templates/components/wds-global-navigation/search-modal.hbs]: add check for deactivateSearch and do not pass empty action to component

File: app/templates/components/wds-global-navigation/search-modal.hbs [hunk 8]
```diff
@@ -0,0 +1,17 @@
+{{#wds-button click=(action 'openModal' 'search') class='wds-global-navigation__modal-control wds-global-navigation__modal-control-search' text=true type='button'}}
+	{{wds-icon name='magnifying-glass'}}
+{{/wds-button}}
+
+{{#if isOpen}}
+	<div class="wds-global-navigation__modal wds-search-modal">
+		<div class="wds-search-modal__form-and-links-wrapper">
+			{{wds-global-navigation/search
+				model=model
+				onSearchToggleClicked=onSearchToggleClicked
```
Comment (hakubo) [inline app/templates/components/wds-global-navigation/search-modal.hbs]: I'd vote for always using (action onSearchToggleClicked) not much difference but I think it shows clearer that a function is passed

**Pull request merged**